pkg/chains/formats/slsa/v2alpha2/internal/build_definitions/taskrun.go

```diff
@@ -0,0 +1,65 @@
+package builddefinitions
+
+import (
+	"fmt"
+
+	"github.com/tektoncd/chains/pkg/chains/objects"
+)
+
+type TaskBuildType struct {
+	BuildType          string
+	Tro                *objects.TaskRunObject
+	InternalParameters func(tro *objects.TaskRunObject) map[string]any
+}
+
+// internalParameters adds the tekton feature flags that were enabled
+// for the taskrun.
+func SLSATaskInternalParameters(tro *objects.TaskRunObject) map[string]any {
+	internalParams := make(map[string]any)
+	if tro.Status.Provenance != nil && tro.Status.Provenance.FeatureFlags != nil {
+		internalParams["tekton-pipelines-feature-flags"] = *tro.Status.Provenance.FeatureFlags
+	}
+	return internalParams
+}
+
+func TektonTaskInternalParameters(tro *objects.TaskRunObject) map[string]any {
```

Comment:
Consider this. Combine `TektonTaskInternalParameters` and `SLSATaskInternalParameters` into a single function which is a method of `TaskBuildType`. `TaskBuildType.BuildType` to determine whether or not the additional information should be included. I think that will simplify these code changes by quite a bit:

```go
type TaskBuildType struct {
	BuildType          string
	Tro                *objects.TaskRunObject
}

func (t *TaskBuildType) GetInternalParameters() map[string]any {
 	internalParams := make(map[string]any)
	if tro.Status.Provenance != nil && tro.Status.Provenance.FeatureFlags != nil {
		internalParams["tekton-pipelines-feature-flags"] = tro.Status.Provenance.FeatureFlags
	}
	if t.BuildType == "..." {
		internalParams["labels"] = tro.ObjectMeta.Labels
		internalParams["annotations"] = tro.ObjectMeta.Annotations
	}
	return internalParams
}
```

I think the small variation can be cleanly handled like that. If we need something more sophisticated, I'd suggest an interface with different implementations instead.

Reply:
I didn't do this since I put the logic around the buildType in `getBuildDefinition` and didn't want to duplicate it. https://github.com/tektoncd/chains/pull/895/files#diff-46036042c702e6cd4cff49ca6f7eae50d0bc874b4b6e20da07eecca5df6a1581R104